### DISCUSSION TITLE
Translation of documents (.doc, .ppt, txt)

### DISCUSSION THREAD

**Discussion by dingedi** [argosopentech/argos-translate#201] [Ideas]
Hello,

I was thinking a bit about a PR to add the possibility of translating documents (doc, ppt, txt) for LibreTranslate.

Would it be better to integrate this functionality:
  - directly in argos-translate
  - in a separate argos package like translate-html
  - directly in LibreTranslate

**Comment by PJ-Finlay** [comment]
Either a separate package like translate-html or directly in LibreTranslate.

Argos Translate has the ITag interface which does translation on abstract html like tags. This is what translate-html combines with an HTML parser to translate HTML. I put it in its own repo to keep an extra dependency out of Argos Translate and so people could use it on its own. Directly integrating into LibreTranslate would work well too.

**Reply by dingedi** [reply]
Alright thanks @PJ-Finlay , I am thinking of doing a separate package like translate-html and then integrating it into libretranslate, it will be a good organization I think.

**Reply by dingedi** [reply]
I started to work on it. For now it works with .docx

[https://github.com/dingedi/argos-translate-files/tree/develop](https://github.com/dingedi/argos-translate-files/tree/develop)

I will integrate it into LibreTranslate when the code will be stable @pierotofy

**Reply by pierotofy** [reply]
That's awesome! I second PJ's opinion, a separate package just like translate-html is the way to go (and you can then pull it into LT).

**Reply by dingedi** [reply]
I am starting to work on the integration with LibreTranslate [https://github.com/LibreTranslate/LibreTranslate/pull/157](https://github.com/LibreTranslate/LibreTranslate/pull/157)
